File: epi_judge_cpp/int_as_array_multiply.cc
#include <algorithm>
#include <vector>
#include "test_framework/generic_test.h"
using namespace std;


// My First Attempt
vector<int> Multiply(vector<int> num1, vector<int> num2) {

  vector<int> ans( num1.size()+ num2.size() );
  const int sign = (num1.front() < 0) ^ (num2.front() < 0) ? -1 : 1;
  num1.front() = abs(num1.front()), num2.front() = abs(num2.front());

  for ( int i = num1.size()-1; i >= 0; --i ) {
    for ( int j = num2.size()-1; j >= 0; --j ) {
      ans[i+j+1] += num1[i]*num2[j];
      ans[i+j] += ans[i+j+1]/10;
      ans[i+j+1] %= 10;
    }
  }

  ans = {
      find_if_not( begin(ans), end(ans), [](int a) { return a == 0; }),
      end(ans)};

  if (ans.empty()) {
    return {0};
  }

  ans.front() *= sign;
  return ans;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"num1", "num2"};
  return GenericTestMain(args, "int_as_array_multiply.cc",
                         "int_as_array_multiply.tsv", &Multiply,
                         DefaultComparator{}, param_names);
}

File: epi_judge_python/int_as_array_multiply.py
from test_framework import generic_test


def multiply(num1, num2):
    sign = -1 if (num1[0] < 0) ^ (num2[0] < 0) else 1
    num1[0], num2[0] = abs(num1[0]), abs(num2[0])

    result = [0] * (len(num1) + len(num2))
    for i in reversed(range(len(num1))):
        for j in reversed(range(len(num2))):
            result[i+j+1] += num1[i] * num2[j]
            result[i+j] += result[i+j+1] // 10
            result[i+j+1] %= 10

    i = 0
    while i < len(result) and result[i] == 0:
        i+= 1

    result = [0] if i == len(result) else result[i:]

    return [sign * result[0]] + result[1:]

# def multiply(num1, num2):
#     sign = -1 if ( num1[0] < 0 ) ^ (num2[0] < 0) else 1
#     num1[0], num2[0] = abs(num1[0]), abs(num2[0])

#     result = [0] * (len(num1) + len(num2))
#     for i in reversed(range(len(num1))):
#         for j in reversed(range(len(num2))):
#             result[i+j+1] += num1[i] * num2[j]
#             result[i+j] += result[i+j+1] // 10
#             result[i+j+1] %= 10
    
#     result = result[next((i for i, x in enumerate(result)
#                          if x != 0), len(result)):] or [0]
    
#     return [sign * result[0]] + result[1:]

if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("int_as_array_multiply.py",
                                       'int_as_array_multiply.tsv', multiply))
